doc/source/guide/in_depth/code/functional_indepth_example.py

```diff
@@ -0,0 +1,124 @@
+import odl
+
+
+# Here we define the functional
+class MyFunctional(odl.solvers.Functional):
+
+    """This is my functional: ``||x||_2^2 + <x, y>``."""
+
+    def __init__(self, space, linear_term):
+        """Initialize a new instance."""
+        # This comand calls the init of Functional and sets a number of
+        # parameters associated with a functional. All but domain have default
+        # values if not set.
+        super().__init__(space=space, linear=False, grad_lipschitz=2)
+
+        # We need to check that linear_term is in the domain. Then we store the
+        # value of linear_term for future use.
+        if linear_term not in space:
```

Comment:
Frankly I'd skip such tests for the example, unless our plan here is to teach good coding practice.